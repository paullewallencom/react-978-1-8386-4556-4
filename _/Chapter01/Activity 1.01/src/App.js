import React, { Component } from "react";

const Header = props => <h1>{props.title}</h1>;

const InventoryItem = props => (
  <div className="InventoryItem">
    <strong>{props.itemName}</strong>
    <hr />
    <p>{props.itemPrice}</p>
  </div>
);

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      items: [
        { itemName: "Shoe", itemPrice: 5 },
        { itemName: "Sock", itemPrice: 3 }
      ]
    };
  }
  render() {
    return (
      <div className="App">
        <Header title="BuyStuff" />
        <InventoryItem
          itemName={this.state.items[0].itemName}
          itemPrice={this.state.items[0].itemPrice}
        />
        <InventoryItem
          itemName={this.state.items[1].itemName}
          itemPrice={this.state.items[1].itemPrice}
        />
      </div>
    );
  }
}

export default App;
