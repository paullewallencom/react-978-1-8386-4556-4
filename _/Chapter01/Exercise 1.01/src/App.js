import React from "react";

class App extends React.Component {
  render() {
    return (
      <div className="Example" id="my-element">
        <div
          className="greeting"
          style={{ background: "black", color: "white" }}
          onClick={() => alert("hello")}
        >
          Hello World
        </div>
      </div>
    );
  }
}

export default App;
