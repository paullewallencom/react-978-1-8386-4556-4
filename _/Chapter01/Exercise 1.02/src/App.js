import React from "react";

const App = props => (
  <div>
    <h1>My App</h1>
    <br />
    Hello React!
  </div>
);

export default App;
