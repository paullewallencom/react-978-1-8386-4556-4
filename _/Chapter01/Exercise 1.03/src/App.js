import React, { Component } from 'react';
 
const Greeting = props => <p>Hello {props.name}!</p>;
 
class App extends Component {
  constructor(props) {
    super(props);
    this.title = 'React App';
    this.state = { clickCounter: 0 };
  }
  renderClickCount() {
    return <p>I've been clicked {this.state.clickCounter} times!</p>;
  }
  render() {
    return (
      <div>
        <h1>{this.title}</h1>
        <br />
        <Greeting name="User" />
        <br />
        {this.renderClickCount()}
      </div>
    );
  }
}
 
export default App;

