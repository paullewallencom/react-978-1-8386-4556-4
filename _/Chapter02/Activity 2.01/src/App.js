import React, { Component } from 'react';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { field: "", submitDisabled: true };
  }
  renderFieldLength() {
    return <p>{`${this.state.field.length} character(s)!`}</p>
  }
  updateFieldLength(event) {
    const field = event.target.value;
    this.setState({ field }, () => {
      this.validateFieldLength();
    });
  }
  validateFieldLength() {
    if (this.state.submitDisabled && this.state.field.length > 100) {
      this.setState({ submitDisabled: false });
    } else if (!this.state.submitDisabled && this.state.field.length <= 100) {
      this.setState({ submitDisabled: true });
    }
  }
  submitForm() {
    alert("Submitting the blog post!");
  }
  render() {
    return (
      <div className="App">
        <h2>Blog Post Writer</h2>
        <hr />
        <strong>Write your post here</strong><br />
        <small>Must be at least 100 characters!</small><br />
        <textarea cols="80" rows="25" onChange={this.updateFieldLength.bind(this)}></textarea>
        <br />
        {this.renderFieldLength()}
        <br />
        <button disabled={this.state.submitDisabled} onClick={this.submitForm}>Submit Post</button>
      </div>
    );
  }
}

export default App;
