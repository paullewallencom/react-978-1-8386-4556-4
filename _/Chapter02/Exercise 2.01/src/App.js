import React, { Component } from 'react';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      passwordConfirmation: '',
      email: '',
      errors: []
    };

  }
  displayForm() {
    return (
      <div>
        Username: <input type="text" /><br />
        Password: <input type="text" /><br />
        Password Confirmation: <input type="text" /><br />
        Email: <input type="text" /><br />
        <br />
        <button>Submit</button>
      </div>
    );
  }
  render() {
    return (
      <div className="App">
        Create Account
      <hr />
        {this.displayForm()}
      </div>
    )
  }

}

export default App;
