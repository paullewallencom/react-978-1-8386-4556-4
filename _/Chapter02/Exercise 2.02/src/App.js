import React, { Component } from 'react';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      passwordConfirmation: '',
      email: '',
      errors: []
    };

  }
  displayForm() {
    return (
      <div>    
        Username: <input type="text" /><br />
        Password: <input type="text" /><br />
        Password Confirmation: <input type="text" /><br />
        Email: <input type="text" /><br />
        <br />
        <button onClick={this.submitForm}>Submit</button> 
      </div>
      
    );
  }
  submitForm(event) {
    console.log("Submitting the form now...");
    console.log(event);
  }

  render() {
    return (
      <div className="App">
        Create Account
      <hr />
        {this.displayForm()}
      </div>
    )
  }

}

export default App;
