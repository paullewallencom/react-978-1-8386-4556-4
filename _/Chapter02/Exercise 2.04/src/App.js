import React, { Component } from "react";
class App extends Component {
  state = {
    username: "",
    password: "",
    passwordConfirmation: "",
    email: "",
    errors: [],
  };
  validateUsernameOnBlur = (event) => {
    console.log("I should validate whatever is in ", event.target.value);
    this.setState();
  };
  displayForm() {
    return (
      <div>
        Username: <input type="text" onBlur={this.validateUsernameOnBlur} />
        <br />
        Password: <input type="text" />
        <br />
        Password Confirmation: <input type="text" />
        <br />
        Email: <input type="text" />
        <br />
        <br />
        <button onClick={this.submitForm}>Submit</button>
      </div>
    );
  }
  submitForm(event) {
    console.log("Submitting the form now...");
    console.log(event);
  }
  render() {
    return (
      <div className="App">
        Create Account
        <hr />
        {this.displayForm()}
      </div>
    );
  }
}
export default App;

