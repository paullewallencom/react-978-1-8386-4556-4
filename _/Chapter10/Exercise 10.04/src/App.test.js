import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from './App';

test('renders correctly', () => {
  const {container} = render(<App/>);
  expect(container).toMatchSnapshot();
});
