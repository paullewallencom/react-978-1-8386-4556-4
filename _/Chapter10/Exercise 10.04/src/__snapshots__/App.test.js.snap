// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="App"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Form
        </a>
      </li>
      <li>
        <a
          href="/Feed"
        >
          Feed
        </a>
      </li>
      <li>
        <a
          href="/Dashboard"
        >
          Dashboard
        </a>
      </li>
    </ul>
    <div>
      <h1>
        Signup
      </h1>
      <form
        action="#"
      >
        <label>
          Email*:
          <input
            name="email"
            value=""
          />
        </label>
        <label>
          Name*:
          <input
            name="username"
            value=""
          />
        </label>
        <button
          type="submit"
        >
          Submit
        </button>
        <span>
          Form is Dirty: 
          False
        </span>
      </form>
    </div>
  </div>
</div>
`;
